src/ports/postgres/modules/graph/hits.py_in

```diff
@@ -0,0 +1,439 @@
+
+                plpy.execute("""
+                        TRUNCATE TABLE {message_unconv_hub};
+                        INSERT INTO {message_unconv_hub}
+                        SELECT {cur}.{vertex_id}
+                        FROM {message}
+                        INNER JOIN {cur}
+                        ON {cur}.{vertex_id}={message}.{vertex_id}
+                        WHERE ABS({cur}.hub-{message}.hub) > {threshold}
+                        {limit}
+                    """.format(**locals()))
+                unconverged_hub_num = plpy.execute("""SELECT COUNT(*) AS cnt FROM {0}
+                    """.format(message_unconv_hub))[0]["cnt"]
+            else:
+                # Do not run convergence test if threshold=0, since that implies
+                # the user wants to run max_iter iterations.
+                unconverged_authority_num = 1
+                unconverged_hub_num = 1
+
+            plpy.execute("DROP TABLE IF EXISTS {0}".format(cur))
+            plpy.execute("""ALTER TABLE {message} RENAME TO {cur}
+                    """.format(**locals()))
+
+            if unconverged_authority_num == 0 and unconverged_hub_num == 0:
+                break
+
+        plpy.execute("""
+            ALTER TABLE {table_name}
+            RENAME TO {out_table}
+            """.format(table_name=cur, **locals()))
+        # If there are nodes that have no incoming edges, they are not
+        # captured in the previous calculation. The authority and hub value should
+        # both be 0 for those nodes. Insert entries for such nodes with authority=0 
+        # and hub=0 into out_table.
+        plpy.execute("""
+                INSERT INTO {out_table}
+                SELECT *
+                FROM {nodes_with_no_incoming_edges}
+            """.format(**locals()))
+        #update summary_table
+        plpy.execute("""
+            INSERT INTO {summary_table} 
+            VALUES({iteration_num}+1)
+            """.format(**locals()))
+
+        # Cleanup All the intermediate tables
+        plpy.execute("""DROP TABLE IF EXISTS {0},{1},{2},{3},{4},{5}
+            """.format(edge_temp_table, cur, message,
+                       message_unconv_authority, message_unconv_hub, nodes_with_no_incoming_edges))
+
+def hits_help(schema_madlib, message, **kwargs):
+    """
+    Help function for hits
+
+    Args:
+        @param schema_madlib
+        @param message: string, Help message string
+        @param kwargs
+
+    Returns:
+        String. Help/usage information
+    """
+    if message is not None and \
+            message.lower() in ("usage", "help", "?"):
+        help_string = "Get from method below"
+        help_string = get_graph_usage(schema_madlib, 'HITS',
+                                      """out_table     TEXT, -- Name of the output table for HITS
+    max_iter      INTEGER, -- Maximum iteration number (DEFAULT = 100)
+    threshold     DOUBLE PRECISION, -- Stopping criteria (DEFAULT = 1/(N*1000),
```

Comment:
Indent an extra tab (4 spaces) to the right, for both `max_iter` and `threshold`.